packages/shared/lib/core/wallet/utils/parseTransactionsToActivities.ts

```diff
@@ -0,0 +1,84 @@
+import { truncateString } from '@lib/helpers'
+import { ActivityAsyncStatus, ActivityDirection, ActivityType, IActivity } from '@core/wallet'
+import { AccountMessage } from '@lib/typings/wallet'
+import { formatUnitBestMatch } from '@lib/units'
+import {
+    findAccountWithAddress,
+    findAccountWithAnyAddress,
+    getIncomingFlag,
+    getInternalFlag,
+    receiverAddressesFromTransactionPayload,
+    sendAddressFromTransactionPayload,
+} from '@lib/wallet'
+import { formatTokenAmountBestMatch } from './formatTokenAmountBestMatch'
+
+export function parseTransactionsToActivities(transactions: AccountMessage[]): IActivity[] {
+    return transactions.map((transaction) => ({
+        id: transaction.id,
+        type: 'Transaction',
+        timestamp: transaction.timestamp,
+        activityType: getActivityType(transaction.payload),
+        internal: getInternalFlag(transaction.payload),
+        direction: getIncomingFlag(transaction.payload) ? ActivityDirection.In : ActivityDirection.Out,
+        confirmed: transaction.confirmed,
+        subjectAccountName: getSubjectName(transaction.payload),
+        subjectAddress: getSubjectAddress(transaction.payload),
+        amount: getMessageValue(transaction.payload),
+        fiatAmount: '-',
+        token: {
+            name: 'Iota',
+            unit: 'i',
+            useMetricPrefix: true,
+        },
+    }))
+}
+
+function getMessageValue(payload) {
+    const token = {
+        name: 'Iota',
+        unit: 'i',
+        useMetricPrefix: true,
+    }
+    return `${!getIncomingFlag(payload) ? '-' : ''}${formatTokenAmountBestMatch(
```

Comment:
I think we should move the formatting of this too the component where it is presented, as we might want to use different formatting in different places, or we might want to use the value of an activity for a calculation and would be better to just store the rawAmount in this store.